src/auth/auth.controller.ts

```diff
@@ -43,21 +43,24 @@ export const hasRole = (allowRoles: string[]) => {
   ) => {
 
     const user = req.user as any
-    let userRole: string | undefined
+    let userRoles: string[] = []
 
     if (user.admin) {
-      userRole = 'ADMIN';
-    } else if (user.doctor) {
-      userRole = 'DOCTOR';
-    } else if (user.patient) {
-      userRole = 'PATIENT';
+      userRoles?.push('ADMIN');
+    }
+    if (user.doctor) {
+      userRoles?.push('DOCTOR');
+    }
+    if (user.patient) {
+      userRoles?.push('PATIENT');
     }
 
-    if (!userRole) {
+    if (userRoles.length === 0) {
```

Comment:
Why not `if (!userRoles.length {...`

Reply:
You are right, for semantic reasons I will modify it.